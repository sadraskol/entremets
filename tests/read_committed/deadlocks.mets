init do
    `insert into accounts (id, balance) values (11, 0), (22, 0)`
end

process do
    transaction tx1 read_committed do
        `update accounts set balance := 100 where id = 11`
        `update accounts set balance := 100 where id = 22`
    end
end

process do
    transaction tx2 read_committed do
        `update accounts set balance := 50 where id = 22`
        `update accounts set balance := 50 where id = 11`
    end
end

property = eventually(`select balance from accounts` in {{50, 50}, {100, 100}})