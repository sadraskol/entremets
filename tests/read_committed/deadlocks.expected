System ran into a deadlock: